fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/HashJobStore.java

```diff
@@ -51,4 +61,43 @@ public HashJobStore() {
     public Observable<JobRow> getAllRows() {
         return Observable.from(store.values());
     }
+
+    @Override
+    public Observable<JobRow> getFilteredRows(Set<ApiJobStoreFilter> apiJobStoreFilters)
+            throws IllegalArgumentException {
+        return Observable.from(
+                store.entrySet().stream()
+                        .filter(entry -> satisfiesFilters(apiJobStoreFilters, entry.getValue()))
+                        .map(entry -> entry.getValue())
+                        .collect(Collectors.toSet())
+        );
+    }
+
+    /**
+     * This method checks if the given JobRow satisfies all the ApiJobStoreFilters and returns true if it does.
+     * If a JobField in any of the filters is not a part the JobRow, this method throws a IllegalArgumentException.
+     *
+     * @param apiJobStoreFilters  A Set of ApiJobStoreFilters specifying the different conditions to be satisfied
+     * @param jobRow  The JobRow which needs to be inspected
+     *
+     * @return true if the JobRow satisfies all the filters, false otherwise
+     *
+     * @throws IllegalArgumentException if a JobField in any of the filters is not a part the JobRow
+     */
+    private boolean satisfiesFilters(Set<ApiJobStoreFilter> apiJobStoreFilters, JobRow jobRow)
+            throws IllegalArgumentException {
+        for (ApiJobStoreFilter filter : apiJobStoreFilters) {
+            JobField jobField = filter.getJobField();
+            if (!jobRow.containsKey(jobField)) {
+                Set<JobField> jobFields = jobRow.keySet();
```

Comment:
I would suggest renaming this to something like `actualJobFields` rather than `jobFields`. It ensures that two variables that represent very different things (the `JobField` to filter on, vs. the `JobFields` actually in the `JobRow`) have names that differ by more than one letter.